org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -1776,4 +1799,29 @@ public static boolean isExcludedFile(List<String> patterns, String uriString) {
 		FileSystem fileSystems = path.getFileSystem();
 		return !patterns.stream().filter(pattern -> fileSystems.getPathMatcher("glob:" + pattern).matches(path)).collect(Collectors.toList()).isEmpty();
 	}
+
+	/**
+	 * Returns whether this member exists in the source.
+	 *
+	 * @param member
+	 *            the member
+	 * @return <code>true</code> if this element exists in the source, and
+	 *         <code>false</code> if this element does not exist
+	 */
+	public static boolean isGenerated(IMember member) {
+		if (member != null && member.exists()) {
+			try {
+				ISourceRange range = member.getNameRange();
+				if (SourceRange.isAvailable(range)) {
+					ICompilationUnit unit = member.getCompilationUnit();

```

Comment:
According to https://github.com/projectlombok/lombok/blob/731bb185077918af8bc1e6a9e6bb538b2d3fbbd8/src/eclipseAgent/lombok/launch/PatchFixesHider.java#L418-L426 , this check should work. If it works in our case, it does seem simpler.

We use the one from https://github.com/eclipse/eclipse.jdt.ls/blob/681c5f7262dc0820614123dbc17a68f81d001000/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InlayHintVisitor.java#L142 to avoid inlay hints in some cases.

Reply:
https://github.com/projectlombok/lombok/blob/731bb185077918af8bc1e6a9e6bb538b2d3fbbd8/src/eclipseAgent/lombok/launch/PatchFixesHider.java#L418-L426 doesn't work. 
` member.getNameRange().equals(member.getSourceRange())` is always true.

Reply:
I tried with the snippet in https://github.com/redhat-developer/vscode-java/issues/2805#issue-1453069967 . Activating go-to-definition on `getC()` and having a breakpoint at `JDTUtils.isGenerated(IMember)`. The comparison is `true`.

Is this not working at all for you ?

Reply:
Yes, it works fine. See https://github.com/eclipse/eclipse.jdt.ls/pull/2339/commits/ee1084a170689bea720fed3437c53cf586ef0d4b#diff-38af31aa96ae8ee1debae74bf388b43dea8891646b1032ecdb2f4a7ea3e06f8cR1808